src/NuGet.Protocol.Catalog/Serialization/BareStringOrArrayConcatenatingConverter.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace NuGet.Protocol.Catalog
+{
+    /// <summary>
+    /// A <see cref="JsonConverter"/> implementation that leaves the bare string values as is, but converts arrays of strings
+    /// into single strings by concatenating all elements.
+    /// 
+    /// Exists to work around Catalog issues like in the following leaf:
+    /// https://api.nuget.org/v3/catalog0/data/2018.03.11.05.06.09/fluentconsoleapplication.0.1.0.json
+    /// (see releaseNotes).
+    /// </summary>
+    public class BareStringOrArrayConcatenatingConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType == typeof(string);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
```

Comment:
If a registration page has one of these, how does client handle it? I think we should try to mimic the behavior of client if it permissive here.

Reply:
https://api.nuget.org/v3/registration3-gz-semver2/fluentconsoleapplication/index.json

Registration doesn't have `releaseNotes`, apparently.

Reply:
Oh, my bad. What about if `description` had this problem? My point is array concatenation seems like an arbitrary decision and we should try to be consistent with anything we already have. 

Also, how does client or client APIs handle such a .nuspec on disk?

Reply:
https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Protocol/Model/PackageSearchMetadata.cs

Looks like it would fail in the same way that this throws.

Reply:
Updated to pick the first.